src/main/java/org/cryptomator/linux/quickaccess/DolphinPlaces.java

```diff
@@ -0,0 +1,161 @@
+import org.cryptomator.integrations.common.Priority;
+import org.cryptomator.integrations.quickaccess.QuickAccessService;
+import org.cryptomator.integrations.quickaccess.QuickAccessServiceException;
+import org.xml.sax.SAXException;
+
+import javax.xml.XMLConstants;
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Implemenation of the {@link QuickAccessService} for KDE desktop environments using Dolphin file browser.
+ */
+@DisplayName("KDE Dolphin Places")
+@OperatingSystem(OperatingSystem.Value.LINUX)
+@Priority(90)
+public class DolphinPlaces implements QuickAccessService {
+
+	private static final int MAX_FILE_SIZE = 1 << 15; //xml is quite verbose
+	private static final Path PLACES_FILE = Path.of(System.getProperty("user.home"), ".local/share/user-places.xbel");
+	private static final Path TMP_FILE = PLACES_FILE.resolveSibling("user-places.xbel.cryptomator.tmp");
+	private static final Lock MODIFY_LOCK = new ReentrantReadWriteLock().writeLock();
+	private static final String ENTRY_TEMPLATE = """
+			<bookmark href="%s">
+			 <title>%s</title>
+			 <info>
+			  <metadata owner="http://freedesktop.org">
+			   <bookmark:icon name="drive-harddisk"/>
+			  </metadata>
+			  <metadata owner="https://cryptomator.org">
+			  	<id>%s</id>
+			  </metadata>
+			 </info>
+			</bookmark>""";
+
+
+	private static final Validator xmlValidator;
+
+	static {
+		SchemaFactory factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+		try (var schemaDefinition = DolphinPlaces.class.getResourceAsStream("/xbel-1.0.xsd")) {
+			Source schemaFile = new StreamSource(schemaDefinition);
+			xmlValidator = factory.newSchema(schemaFile).newValidator();
+		} catch (IOException | SAXException e) {
+			throw new IllegalStateException("Failed to load included XBEL schema definition file.", e);
+		}
+	}
+
+
+	@Override
+	public QuickAccessService.QuickAccessEntry add(Path target, String displayName) throws QuickAccessServiceException {
+		String id = UUID.randomUUID().toString();
+		try {
+			MODIFY_LOCK.lock();
+			if (Files.size(PLACES_FILE) > MAX_FILE_SIZE) {
+				throw new IOException("File %s exceeds size of %d bytes".formatted(PLACES_FILE, MAX_FILE_SIZE));
+			}
+			var placesContent = Files.readString(PLACES_FILE);
+			//validate
+			xmlValidator.validate(new StreamSource(new StringReader(placesContent)));
+			// modify
+			int insertIndex = placesContent.lastIndexOf("</xbel>"); //cannot be -1 due to validation
```

Comment:
As far as I know XML collapses whitespaces, i.e. `</xbel>` and `< /xbel >` are equal as far as the validator is concerned. This may cause serious trouble here (and to a lesser extend in a few other places.)

Reply:
By the [xml standard](https://www.w3.org/TR/REC-xml/#sec-starttags), start and end tags may only contain whitespaces after the tag name. Therefore, i switched to only match tag start+ tag name in https://github.com/cryptomator/integrations-linux/pull/85/commits/5e9eacc5cbd51c4dc3b1ea8a9d5734ad591c49fb